plugins/brimcap/brimcap-plugin.ts

```diff
@@ -0,0 +1,312 @@
+import {DateTimeFormatter, LocalDateTime, ZoneOffset} from "@js-joda/core"
+import path, {join} from "path"
+import {fetchCorrelation} from "../../ppl/detail/flows/fetch"
+import open from "../../src/js/lib/open"
+import {AppDispatch} from "../../src/js/state/types"
+import {zng} from "../../zealot"
+import {Record} from "../../zealot/zng"
+import BrimcapCLI, {searchOptions} from "./brimcap-cli"
+import BrimApi from "../../src/js/api"
+import {IngestParams} from "../../src/js/brim/ingest/getParams"
+import fsExtra, {pathExistsSync} from "fs-extra"
+import errors from "src/js/errors"
+import {reactElementProps} from "../../src/js/test/integration"
+
+export default class BrimcapPlugin {
+  private cli: BrimcapCLI
+  // currentData represents the data detail currently seen in the Brim detail pane/window
+  private currentConn = null
+  // selectedData represents the data detail currently selected and highlighted in the search viewer
+  private selectedConn = null
+  private cleanupFns: Function[] = []
+  private brimcapDataRoot = ""
+  private brimcapBinPath = ""
+  private toastConfig = {
+    loading: {
+      // setTimeout's maximum value is a 32-bit int, so we explicitly specify here
+      // also, once https://github.com/timolins/react-hot-toast/pull/37 merges, we can set this to -1
+      duration: 2 ** 31 - 1
+    },
+    success: {
+      duration: 3000
+    },
+    error: {
+      duration: 5000
+    }
+  }
+
+  constructor(private api: BrimApi) {
+    // RFC: what interface do we want the app to provide for this sort of data?
+    const {dataRoot, zdepsDirectory} = api.getAppConfig()
+
+    const commandName = process.platform === "win32" ? "brimcap.exe" : "brimcap"
+    this.brimcapBinPath = path.join(zdepsDirectory, commandName)
+    this.cli = new BrimcapCLI(this.brimcapBinPath)
+
+    this.brimcapDataRoot = path.join(dataRoot, "brimcap-root")
+  }
+
+  public init() {
+    fsExtra.ensureDirSync(this.brimcapDataRoot)
+
+    if (!pathExistsSync(this.brimcapBinPath)) {
+      console.error(
+        "The brimcap plugin requires the 'brimcap' CLI tool to also be installed. Please download it here: https://github.com/brimdata/brimcap/releases"
```

Comment:
Nit: This message is no longer quite right now that we're bundling brimcap. Since the error now indicates a packaging problem, maybe suggest that the user create a brimdata/brim issue or otherwise reach out for help?